includes/class-cli.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * CLI Support for Press Sync.
+ *
+ * @since 0.1.0
+ */
+class Press_Sync_CLI {
+
+	/**
+	 * Parent plugin class.
+	 *
+	 * @var   Press_Sync
+	 * @since 0.1.0
+	 */
+	protected $plugin = null;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Press_Sync $plugin The Press Sync plugin.
+	 */
+	public function __construct( Press_Sync $plugin ) {
+
+		$this->plugin = $plugin;
+
+		if ( ! class_exists( 'WP_CLI' ) ) {
+			return;
+		}
+
+		// Register the CLI Commands.
+		WP_CLI::add_command( 'press-sync posts', array( $this, 'sync_posts' ) );
+		WP_CLI::add_command( 'press-sync media', array( $this, 'sync_media' ) );
+		WP_CLI::add_command( 'press-sync pages', array( $this, 'sync_pages' ) );
+		WP_CLI::add_command( 'press-sync users', array( $this, 'sync_users' ) );
+	}
+
+	/**
+	 * Synchronize posts.
+	 *
+	 * @param array $args       The arguments.
+	 * @param array $assoc_args The associative arugments.
+	 *
+	 * @synopsis --remote_domain=<remote_domain> --remote_press_sync_key=<remote_press_sync_key>
+	 */
+	public function sync_posts( $args, $assoc_args ) {
+
+		$response = $this->plugin->sync_content( 'post', $assoc_args, true );
+
+		$this->return_response( $response );
+	}
+
+	/**
+	 * Synchronize media.
+	 *
+	 * @param array $args       The arguments.
+	 * @param array $assoc_args The associative arugments.
+	 *
+	 * @synopsis --remote_domain=<remote_domain> --remote_press_sync_key=<remote_press_sync_key>
+	 */
+	public function sync_media( $args, $assoc_args ) {
+
+		$response = $this->plugin->sync_content( 'media', $assoc_args );
+
+		$this->return_response( $response );
+	}
+
+	/**
+	 * Synchronize pages.
+	 *
+	 * @param array $args       The arguments.
+	 * @param array $assoc_args The associative arugments.
+	 *
+	 * @synopsis --remote_domain=<remote_domain> --remote_press_sync_key=<remote_press_sync_key>
+	 */
+	public function sync_pages( $args, $assoc_args ) {
+
+		$response = $this->plugin->sync_content( 'page', $assoc_args );
+
+		$this->return_response( $response );
+	}
+
+	/**
+	 * Synchronize users.
+	 *
+	 * @param array $args       The arguments.
+	 * @param array $assoc_args The associative arugments.
+	 *
+	 * @synopsis --remote_domain=<remote_domain> --remote_press_sync_key=<remote_press_sync_key>
+	 */
+	public function sync_users( $args, $assoc_args ) {
+
+		$response = $this->plugin->sync_content( 'user', $assoc_args );
+
+		$this->return_response( $response );
+	}
+
+	/**
+	 * Return the response after synchronizing content.
+	 *
+	 * @param array $response The response after synchronization.
+	 */
+	private function return_response( $response = array() ) {
+
+		$total_objects            = isset( $response['total_objects'] ) ? (int) $response['total_objects'] : 0;
```

Comment:
Definitely not something that needs to be addressed right now, but we'll be working in PHP 7.x and there's a new shorthand operator for `isset`:

```
php > $a = [];
php > $number = (int)($a['somekey'] ?? 0);
php > var_dump($number);
int(0)
php > $a['somekey'] = "55";
php > $number = (int)($a['somekey'] ?? 0);
php > var_dump($number);
int(55)
```

So this could be simplified to

```
$total_objects = (int) ($response['total_objects'] ?? 0);
// etc
```

Note that since there's a cast to `(int)` we do need to wrap the following part of the line, in most cases you would see it as

```
$something = $actuallyNotSet ?? 'default value';
```

Reply:
This is a great recommendation but since this project isn't private, backwards compatibility must be maintained. But these notes are great if/when the project is forked for an environment that forces PHP 7.*